### PR TITLE
Add GRU4Rec example

### PR DESCRIPTION
**Pull request by abheesht17**
Note: The loss in the TFRS example does not converge. So, changed a few things.

### CODE REVIEW

File: examples/sequential_retrieval.py [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+neural network to model the sequential relationship. For more details,
+please refer to the [GRU4Rec](https://arxiv.org/abs/1511.06939) paper.
+
+Let's begin by importing all the necessary libraries, and setting the
+random seed for reproducibility.
+"""
+
+import os
+
+os.environ["KERAS_BACKEND"] = "jax"
```
Comment (hertschuh) [inline examples/sequential_retrieval.py:17]: Any reason to do that? At some point I was thinking we can write tests to run these automatically with each backend. But I never got around to doing that. But for it to work, the backend has to not be set within the file.

Did you test with TF and Torch too?

File: examples/sequential_retrieval.py [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+import random
+
+import keras
+import pandas as pd
+import tensorflow as tf  # Needed only for the dataset
+from keras import ops
+
+import keras_rs
+
+keras.utils.set_random_seed(42)
```
Comment (hertschuh) [inline examples/sequential_retrieval.py:29]: Any reason to do that?

File: examples/sequential_retrieval.py [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+LEARNING_RATE = 0.05
+
+"""
+## Dataset
+
+Next, we need to prepare our dataset. Like we did in the
+[basic retrieval](https://github.com/keras-team/keras-rs/blob/main/examples/basic_retrieval.py)
+example, we are going to use the MovieLens dataset. 
+
+The dataset preparation step is fairly involved. The original ratings dataset
```
Comment (hertschuh) [inline examples/sequential_retrieval.py:69]: Thanks for porting all the dataset massaging stuff here. That was most of the work I imagine.

File: examples/sequential_retrieval.py [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+
+os.environ["KERAS_BACKEND"] = "jax"
+
+import collections
+import random
+
+import keras
+import pandas as pd
+import tensorflow as tf  # Needed only for the dataset
+from keras import ops
```
Comment (hertschuh) [inline examples/sequential_retrieval.py:25]: Further down, you're sometimes using `keras.ops.x` and sometimes `ops.x`. Make it consistent. I don't think this import is necessary.

**Review by hertschuh — COMMENTED** [review]
Great work! There was a lot to do with the data processing.

File: examples/sequential_retrieval.py (hunk 2 above)
Comment (abheesht17) [reply examples/sequential_retrieval.py:29]: I set the seed when I was testing training with different hyperparameters, etc. Can remove/keep it, doesn't matter which way

File: examples/sequential_retrieval.py (hunk 3 above)
Comment (abheesht17) [reply examples/sequential_retrieval.py:69]: Yeah, pretty much. I was trying to not do the conversion to Pandas DF, and somehow use [`tf.data.group_by_window`](https://www.tensorflow.org/api_docs/python/tf/data/Dataset#group_by_window). At some point, I gave up xD

File: examples/sequential_retrieval.py (hunk 1 above)
Comment (abheesht17) [reply examples/sequential_retrieval.py:17]: Yep, working for all backends!

**Review by hertschuh — APPROVED** [review]
